lib/sycamore/sycamore/llms/openai.py

```diff
@@ -462,6 +464,6 @@ async def _generate_awaitable_using_openai_structured(self, prompt_kwargs, llm_k
 
     def _generate_using_guidance(self, prompt_kwargs) -> str:
```

Comment:
Would the next step be to convert `generate_with_openai` to also accept a SimplePrompt and execute that? Then eventually we could deprecate `generate_with_guidance`, if indeed we intend to do that?

Reply:
I think so yeah, we should likely add a flag to `generate` that's more explicit about what you're executing with